lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -120,8 +120,57 @@ namespace
         }
         return TfToken();
     }
+
+    //! \brief  Populate Rprims into the Hydra selection from the UFE scene item.
+    void PopulateSelection(
+        const Ufe::SceneItem::Ptr& item,
+        const Ufe::Path& proxyPath,
+        UsdImagingDelegate& sceneDelegate,
+        const HdSelectionSharedPtr& result)
+    {
+        if (item->runTimeId() != USD_UFE_RUNTIME_ID) {
+            return;
+        }
+
+        const Ufe::Path::Segments& segments = item->path().getSegments();
+        if ((segments.size() != 2) || (proxyPath != segments[0])) {
+            return;
+        }
+
+        SdfPath usdPath(segments[1].string());
```

Comment:
At this point, because of lines 131-133, we know the Ufe::SceneItem is a UsdSceneItem, so we can dynamic_pointer_cast to that and simply call prim() on it.  In fact, dynamic_pointer_cast and item->runTimeId() != USD_UFE_RUNTIME_ID serve exactly the same purpose, so I would replace lines 131-133 with the dynamic_pointer_cast.

Reply:
@ppt-adsk  The new code looks nicer. However, we need to filter the UFE scene items which don't belong to the current proxy shape, see Line 135. Any suggestion how to handle this logic with the new code?

Reply:
Changing a Ufe::SceneItem::Ptr to a UsdSceneItem::Ptr wouldn't change lines 135-138, it would only make line 140 less awkward, since you could ask the UsdPrim directly for its SdfPath without going through a string.

However, lines 135-138 might be more elegant as 

if (!item->path().startsWith(proxyPath)) {
    return;
}

if I'm not mistaken.

Reply:
Great! Thanks.